datadotworld/datadotworld.py

```diff
@@ -0,0 +1,249 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing
+permissions and limitations under the License.
+
+This product includes software developed at data.world, Inc.(http://www.data.world/).
+"""
+from __future__ import absolute_import
+
+import shutil
+import weakref
+from os import path
+from warnings import warn
+
+import requests
+
+from datadotworld.client.api import RestApiClient, RestApiError
+from datadotworld.config import Config
+from datadotworld.models.dataset import LocalDataset
+from datadotworld.models.query import QueryResults
+from datadotworld.util import _user_agent, parse_dataset_key
+
+
+class DataDotWorld(object):
+    """Facade with main features of datadotworld package
+
+    .. note:: In most cases, directly instantiating this class is unnecessary.
+              All functions are conveniently wrapped and exposed at the `datadotworld` package level.
+
+    Parameters
+    ----------
+    profile : str, optional
+        Configuration profile (account) to use
+
+    Attributes
+    ----------
+    api_client
+        REST API client object
+    """
+    def __init__(self, profile='default', **kwargs):
+        # Overrides, for testing
+        self._config = kwargs.get('config', Config(profile))
+
+        self._protocol = 'https'
+        self._query_host = 'query.data.world'
+        self._download_host = 'download.data.world'
+        self.api_client = RestApiClient(self._config)
+
+    def query(self, dataset_key, query, query_type="sql"):
+        """Query an existing dataset
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id or of a url
+        query : str
+            SQL or SPARQL query
+        query_type : {'sql', 'sparql'}, optional
+            The type of the query. Must be either 'sql' or 'sparql'.
+
+        Returns
+        -------
+        Results
+            Object containing the results of the query
+
+        Raises
+        ------
+        RuntimeError
+            If a server error occurs
+        """
+        # TODO Move network request to RestApiClient
+        owner_id, dataset_id = parse_dataset_key(dataset_key)
+        params = {
+            "query": query
+        }
+        url = "{0}://{1}/{2}/{3}/{4}".format(self._protocol, self._query_host, query_type, owner_id, dataset_id)
+        headers = {
+            'User-Agent': _user_agent(),
+            'Accept': 'text/csv',
+            'Authorization': 'Bearer {0}'.format(self._config.auth_token)
+        }
+        response = requests.get(url, params=params, headers=headers)
+        if response.status_code == 200:
+            return QueryResults(response.text)
+        raise RuntimeError('Error executing query: {}'.format(response.text))
+
+    def load_dataset(self, dataset_key, force_update=False):
```

Comment:
I'd like an optional argument to override the destination directory.  Doc that the default goes to `~/.dw/cache` (or the cache_dir setting in the config file, unless it's not configurable?).